src/components/switch/OcSwitch.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+  <span class="oc-switch">
+    <button
+      data-test-id="oc-switch-btn"
+      class="oc-switch-btn"
+      role="switch"
+      :aria-checked="state.toString()"
+      :aria-labelledby="labelId"
+      @click="toggle"
+    />
+    <span :id="labelId" v-text="label" />
```

Comment:
What do you think about using an actual `<label>` instead of a `<span>` as that's already the native html element that fulfills what you want to achieve? 🤔 To keep it consistent with our other form like components I'd prefer to have the label before the button (see e.g. `OcTextInput`).

Reply:
It's not used on purpose as label is not made for buttons

![image](https://user-images.githubusercontent.com/25989331/123398049-653fd600-d5a3-11eb-9469-d12d5ba37747.png)

Reply:
Took a **great** inspiration here from Marcus https://github.com/marcus-herrmann/vue-switch-button/blob/master/src/components/SwitchButton.vue